File: 205-isomorphic-strings/205-isomorphic-strings.cs
public class Solution {
    public bool IsIsomorphic(string str1, string str2) {
        Dictionary<char, char> charCount
            = new Dictionary<char, char>();
        char c = 'a';
        for (int i = 0; i < str1.Length; i++) {
            if (charCount.ContainsKey(str1[i])
                && charCount.TryGetValue(str1[i], out c)) {
                if (c != str2[i])
                    return false;
            }
            else if (!charCount.ContainsValue(str2[i])) {
                charCount.Add(str1[i], str2[i]);
            }
            else {
                return false;
            }
        }
        return true;
    }
}

File: 205-isomorphic-strings/205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length()){
            return false;
        }
        HashMap<Character, Character> map= new HashMap<>();
        for(int i=0; i < s.length(); i++){
            if(map.containsKey(s.charAt(i))){
                if(t.charAt(i) != map.get(s.charAt(i)))
                    return false;
            }
            else if(!map.containsValue(t.charAt(i))){
                map.put(s.charAt(i), t.charAt(i));
            }
            else
                return false;
        }
        return true;
    }
}